ts/features/itwallet/walletInstance/hook/useItwWalletInstanceRevocationAlert.ts

```diff
@@ -0,0 +1,120 @@
+import { Alert, AlertButton, Linking } from "react-native";
+import React from "react";
+import { IOToast } from "@pagopa/io-app-design-system";
+import I18n from "../../../../i18n";
+import {
+  WalletInstanceRevocationReason,
+  WalletInstanceStatus
+} from "../../common/utils/itwTypesUtils";
+import { useIODispatch, useIOSelector } from "../../../../store/hooks";
+import { itwWalletInstanceAlertShownSelector } from "../store/selectors";
+import { itwWalletInstanceSetAlertShown } from "../store/actions";
+
+const closeButtonText = I18n.t(
+  "features.itWallet.walletInstanceRevoked.alert.closeButton"
+);
+const alertCtaText = I18n.t(
+  "features.itWallet.walletInstanceRevoked.alert.cta"
+);
+
+const itwMinIntegrityReqUrl = "https://io.italia.it/documenti-su-io/faq/#n1_12";
+const itwDocsOnIOMultipleDevicesUrl =
+  "https://io.italia.it/documenti-su-io/faq/#n1_14";
+
+/**
+ * Hook to monitor wallet instance status and display alerts if revoked.
+ * @param walletInstanceStatus - The status of the wallet instance, including whether it is revoked and the reason for revocation.
+ */
+export const useItwWalletInstanceRevocationAlert = (
+  walletInstanceStatus: WalletInstanceStatus | undefined
+) => {
+  const dispatch = useIODispatch();
+  const alertShown = useIOSelector(itwWalletInstanceAlertShownSelector);
+
+  React.useEffect(() => {
+    if (walletInstanceStatus?.is_revoked && !alertShown) {
+      showWalletRevocationAlert(walletInstanceStatus.revocation_reason);
+      dispatch(itwWalletInstanceSetAlertShown(true));
+    }
+  }, [walletInstanceStatus, alertShown, dispatch]);
+};
```

Comment:
Just a couple of observations:
- we are in a hook, we could get the `walletInstanceStatus` directly in here, avoiding passing it by parameter. 
- instead of using a stored flag, we can show the alert once based on the `is_revoked` property using a `useOnFirstRender`.

So the hook may look like this:

```suggestion
export const useItwWalletInstanceRevocationAlert = () => {
  const dispatch = useIODispatch();
  const walletInstanceStatus = useIOSelector(itwWalletInstanceStatusSelector);

  useOnFirstRender(
    useCallback(() => {
      if (walletInstanceStatus?.is_revoked) {
        showWalletRevocationAlert(walletInstanceStatus.revocation_reason);
        dispatch(itwWalletInstanceSetAlertShown(true));
      }
    }, [walletInstanceStatus, dispatch])
  );
};
```

A lot cleaner and without having the need to use an extra flag. 
What do you think?

Reply:
Just an update: I noticed that with the proposed implementation, if the user does not navigate to the wallet the first time he logs in, the next time we could lose the information about the revocation reason and lose the ability to show the alert to him.
A solution may be persisting the `status` and then remove it right after the alert display with `itwUpdateWalletInstanceStatus(undefined)`